### PR TITLE
Add examples for flags with Toolbar

### PR DESCRIPTION
**Pull request by AndyBitz**
### Description

Adds examples for LauchDarkly, Optimizely, Split, and Statsig to be used with the Toolbar.

### Demo URL

See all attached Toolbar Flags deployments.

### Type of Change

- [x] New Example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### CODE REVIEW

File: toolbar-flags/launchdarkly-starter/app/.well-known/vercel/flags/route.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import { type NextRequest, NextResponse } from "next/server";
+import { type ApiData, verifyAccess } from "@vercel/flags";
+
+export async function GET(request: NextRequest) {
+  const access = await verifyAccess(request.headers.get("Authorization"));
+  if (!access) return NextResponse.json(null, { status: 401 });
+
+  const data = await getLaunchDarklyData({
+    apiKey: process.env.LAUNCHDARKLY_API_KEY!,
+    environment: "production",
```
Comment (dferber90) [inline toolbar-flags/launchdarkly-starter/app/.well-known/vercel/flags/route.ts:11]: these could be env vars

File: toolbar-flags/launchdarkly-starter/app/page.tsx [hunk 2]
```diff
@@ -0,0 +1,196 @@
+"use server";
```
Comment (dferber90) [inline toolbar-flags/launchdarkly-starter/app/page.tsx:1]: this should be removed, i added it accidentally

File: toolbar-flags/launchdarkly-starter/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    docs: overrides?.docs ?? (await client.variation("docs", context, false)),
+    learn:
+      overrides?.learn ?? (await client.variation("learn", context, false)),
+    templates:
+      overrides?.templates ??
+      (await client.variation("templates", context, false)),
+    deploy:
+      overrides?.deploy ?? (await client.variation("deploy", context, false)),
+  };
+  registerFlagValues(flags);
```
Comment (dferber90) [inline toolbar-flags/launchdarkly-starter/app/page.tsx:52]: let's remove this part

File: toolbar-flags/split-starter/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  children: React.ReactNode;
+}>) {
+  const flags = { instructions: true };
+
+  return (
+    <html lang="en">
+      {/* 
+        This would instruct the toolbar to set plaintext flag overrides
+        <head>
+          <meta name="vercel-flags-plaintext-overrides" />
```
Comment (dferber90) [inline toolbar-flags/split-starter/app/layout.tsx:36]: let's remove this comment as we're now advocating for `overrideEncryptionMode`

File: toolbar-flags/split-starter/app/page.tsx [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+  const key = 'user_123abc';
+
+  const flags = {
+    docs: overrides?.docs ?? client.getTreatment(key, 'docs'),
+    learn: overrides?.learn ?? client.getTreatment(key, 'learn'),
+    templates: overrides?.templates ?? client.getTreatment(key, 'templates'),
+    deploy: overrides?.deploy ?? client.getTreatment(key, 'deploy'),
+  };
+  registerFlagValues(flags);
```
Comment (dferber90) [inline toolbar-flags/split-starter/app/page.tsx:53]: let's delete

File: toolbar-flags/optimizely-starter/app/page.tsx [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+    // @ts-ignore
+    docs: overrides?.docs?.variationKey ?? context.decide('docs').variationKey,
+    // @ts-ignore
+    learn: overrides?.learn?.variationKey ?? context.decide('learn').variationKey,
+    // @ts-ignore
+    templates: overrides?.templates?.variationKey ?? context.decide('templates').variationKey,
+    // @ts-ignore
+    deploy: overrides?.deploy?.variationKey ?? context.decide('deploy').variationKey,
+  };
+  registerFlagValues(flags);
```
Comment (dferber90) [inline toolbar-flags/optimizely-starter/app/page.tsx:55]: let's remove

File: toolbar-flags/statsig-starter/app/page.tsx [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+    // We fall back to false here, but you could fall back to values of your
+    // flag provider instead
+    docs: overrides?.docs ?? Statsig.checkGateSync(user, "docs"),
+    learn: overrides?.learn ?? Statsig.checkGateSync(user, "learn"),
+    templates: overrides?.templates ?? Statsig.checkGateSync(user, "templates"),
+    deploy: overrides?.deploy ?? Statsig.checkGateSync(user, "deploy"),
+    // @ts-ignore
+    get_started: overrides?.get_started?.show ?? Statsig.getExperimentSync(user, "get_started").getValue('show')
+  };
+  registerFlagValues(flags);
```
Comment (dferber90) [inline toolbar-flags/statsig-starter/app/page.tsx:50]: let's remove

**Review by dferber90 — CHANGES_REQUESTED** [review]
<picture data-single-emoji=":excited:" title=":excited:">:excited:</picture>

**Review by leerob — APPROVED** [review]
You might want the versions for `next` at `latest`, but not blocking!

**Pull request merged**